lib/index.js

```diff
@@ -96,16 +106,40 @@ function Socket (type, opts) {
             break;
     }
 
-    this.binding = nn.Socket(this.af_domain, this.protocol);
-    this.queue = [];
+  this.binding = nn.Socket(this.af_domain, this.protocol);
+  this.queue = [];
 
-    for(var sokopt in sol){
-        if(opts.hasOwnProperty(sokopt)) this[sokopt](opts[sokopt]);
-    }
+  /* subscription filters */
+  this.channels = {};
 
-    if(this.af_domain == nn.AF_SP) {
-        if (this.receiver) this._startPollReceive();
+  /* subscription handling at initialization */
+  if (opts.hasOwnProperty('chan')) {
+    if (typeof opts.chan === 'string') {
+      opts.chan.split(',').map(function(i){
+        return i.trim();
+      }).forEach(subscriber.bind(this))
+    } else if (Array.isArray(opts.chan)) {
+      opts.chan.forEach(subscriber.bind(this))
```

Comment:
in both cases, we should map trim, then filter empty elements.  This prevents `chan: ['' ' a', 'b ', ' c ']` and `chan: ", a,b , c "`.